### PR TITLE
Insert BlueR Plus Configuration 

### PR DESCRIPTION
**Pull request by ils15**
Added Marlin Configuration for BlueR Plus

Need testing because i dont have the printer 

What i need to test

- [x] - Motor Direction and steps
- [x] - BLTOUCH Offset and probe procedure (G29)
- [x] - SD Card Read
- [x] - Screen Calibration
- [ ] - PID parameters for hotend and bed (200/60 - 240/100)

Please send feedback here to make adjust before merging

A TIP

You need compile firmware using env:mks_robin_nano35 and rename bin file to Robin_nano43.bin before put it inside SD Card

### CODE REVIEW

**Comment by Snapperjagger** [comment]
It's Rudy Lenglet!
The screen is freezing on the progress bar when tft updating

**Comment by ils15** [comment]
> It's Rudy Lenglet!
> The screen is freezing on the progress bar when tft updating

in general, problem is original sd card
try to format sd card in fat32 and try again

**Comment by Snapperjagger** [comment]
it's the original, provide with the printer. let me try with another one

**Comment by Snapperjagger** [comment]
sorry it was diner time lol!
So it works but the screen is upside down. it needs to flip 180°. 
i got a message: "EEPROM version error  ---   initialize EEPROM"

**Comment by ils15** [comment]
> sorry it was diner time lol!
> So it works but the screen is upside down. it needs to flip 180°.
> i got a message: "EEPROM version error --- initialize EEPROM"

great
i will update screen 
eeprom error is normal when you install a new firmware with new features 
just press green button
I enable 
#define TFT_ROTATION TFT_ROTATE_180 in line 2431
screen will be fixed now

**Comment by Snapperjagger** [comment]
ok
got the ui from Marlin. Temperature probes seems to work

**Comment by ils15** [comment]
> ok
> got the ui from Marlin. Temperature probes seems to work

please check motor moviment too

**Comment by Snapperjagger** [comment]
hotend fan work but it's a difficult to adjust the speed.
the icons showed on the screen don't work but if you touch the screen on the bottom (where the icons should be) i succeed to make the fan run.

**Comment by ils15** [comment]
> hotend fan work but it's a difficult to adjust the speed.
> the icons showed on the screen don't work but if you touch the screen on the bottom (where the icons should be) i succeed to make the fan run.

great
do you fix screen orientation?

i believe we need some calibration initially. I will disable calibration values and lets people calibrate screen for the first time
after you rotate screen, can calibrate it
long press screen and a calibration menu will appear

**Comment by Snapperjagger** [comment]
Hotend heating works too
i don't fix the screen orientation yet

**Comment by ils15** [comment]
> Hotend heating works too
> i don't fix the screen orientation yet
find this line in config file
#define TFT_ROTATION TFT_NO_ROTATION
uncomment and change rotation to
#define TFT_ROTATION TFT_ROTATE_180

**Comment by Snapperjagger** [comment]
yep i just have compiled it with the modification
i try now

**Comment by Snapperjagger** [comment]
i re format the sdcard
it didn't take the update

**Comment by Snapperjagger** [comment]
ok screen is in the right position.
now calibration failed

**Comment by ils15** [comment]
Try again slowly....
I don't know right calibration parameters for this screen

**Comment by Snapperjagger** [comment]
ok good news the axis runs and in the right direction

**Comment by Snapperjagger** [comment]
but i didn't succeed the calibrate the screen. each time i enter into the calibration mode, i don't have enought time to relaese the screen and the top left cross disappear from the screen and pas to the following cross

**Comment by ils15** [comment]
> but i didn't succeed the calibrate the screen. each time i enter into the calibration mode, i don't have enought time to relaese the screen and the top left cross disappear from the screen and pas to the following cross


https://user-images.githubusercontent.com/24917659/103817652-d74c4f00-5045-11eb-960f-07b43d48540f.mp4

**Comment by Snapperjagger** [comment]
so i don't succeed to acceed correctly to certain functions

**Comment by ils15** [comment]
> so i don't succeed to acceed correctly to certain functions

put a gcode in sdcard with a M995 value
put it to print

You can comment these lines in firmware too
  // XPT2046 Touch Screen calibration
  //#define XPT2046_X_CALIBRATION          17880
  //#define XPT2046_Y_CALIBRATION         -12234
  //#define XPT2046_X_OFFSET              -45
  //#define XPT2046_Y_OFFSET               349
  //#define TOUCH_ORIENTATION TOUCH_LANDSCAPE
  after it, printer will run a screen calibration in a first boot

**Comment by Snapperjagger** [comment]
in the configuration.h??

**Comment by ils15** [comment]
> > so i don't succeed to acceed correctly to certain functions
> 
> put a gcode in sdcard with a M995 value
> put it to print



> in the configuration.h??

yes

**Comment by Snapperjagger** [comment]
ok let me see

**Comment by Snapperjagger** [comment]
it don't work know, i have to stay push on the screen to enter in calibration mode then i have the cros like in first time but now after press the last cross i have the top left and the bottom right crosses in same time and the screen is freezed to this status.

**Comment by Snapperjagger** [comment]
i will try by a terminal to enter the M995 code

**Comment by Snapperjagger** [comment]
calibration completed by the terminal

**Comment by ils15** [comment]
> calibration completed by the terminal

great..
Check sd card funcionality please
some users talks about bugs in process

**Comment by Snapperjagger** [comment]
i don't see the jerk parameters in the advance settings

**Comment by Snapperjagger** [comment]
just find a solution for the screen calibration otherwise it seems good.
is it possible now to put the Makerbase UI?? by placing the folders pic and font?

**Comment by ils15** [comment]
> just find a solution for the screen calibration otherwise it seems good.
> is it possible now to put the Makerbase UI?? by placing the folders pic and font?

yes
you define UI here
//#define TFT_CLASSIC_UI
//#define TFT_COLOR_UI
//#define TFT_LVGL_UI

for MKS, need to put assets folder together in folder
i believe you will need to disable some resources in firmware
first compile changing Ui and after it try to debug?

about color UI 
everything working fine?
SD card reading?
About jerk
now marlin uses junction deviation as default
you can change it enabling classic jerk in firmware if you want

**Comment by Snapperjagger** [comment]
Color UI is like the lcd 12486. color are nice
Fan speed variation is ok, sensitivity is a bit hazardous.
The heating bed just turn failed during heating."Heating failed, system stopped! Heater_ID: bed - Printer halted. kill() called!" Maybe an autotune to do
Homing works well
Calibration needs to be done each restart

**Comment by Snapperjagger** [comment]
it detect sdcard but empty

**Comment by ils15** [comment]
> Color UI is like the lcd 12486. color are nice
> Fan speed variation is ok, sensitivity is a bit hazardous.
> The heating bed just turn failed during heating."Heating failed, system stopped! Heater_ID: bed - Printer halted. kill() called!" Maybe an autotune to do
> Homing works well
> Calibration needs to be done each restart

save calibration with a M500
for bed, please run a PID autotune and send values here to replace in firmware 
i will update configuration file
now we need to check if sd card is reading gcode as well

**Comment by Snapperjagger** [comment]
wait i have copy the stl instaed of gcode

**Comment by Snapperjagger** [comment]
Recv: Touch screen calibration completed
Recv: TOUCH_CALIBRATION_X 16128
Recv: TOUCH_CALIBRATION_Y 10841
Recv: TOUCH_OFFSET_X -13
Recv: TOUCH_OFFSET_Y -36
Recv: TOUCH_ORIENTATION TOUCH_PORTRAIT

**Comment by Snapperjagger** [comment]
sdcard is readed

**Comment by Snapperjagger** [comment]
i see my gcode files on the screen

**Comment by Snapperjagger** [comment]
got the error message from octoprint:
Changing monitoring state from "Operational" to "Offline (Error: SerialException: 'device reports readiness to read but returned no data (device disconnected or multiple access on port?)' @ comm.py:_readline:3831)"
Connection closed, closing down monitor
Changing monitoring state from "Offline" to "Detecting serial connection"
Performing autodetection with 1 port/baudrate candidates: /dev/ttyUSB0@250000
Trying port /dev/ttyUSB0, baudrate 250000
Connecting to port /dev/ttyUSB0, baudrate 250000
Handshake attempt #1 with timeout 2.0s
Connected to: Serial<id=0x6a47d610, open=True>(port='/dev/ttyUSB0', baudrate=250000, bytesize=8, parity='N', stopbits=1, timeout=2.0, xonxoff=False, rtscts=False, dsrdtr=False), starting monitor

**Comment by ils15** [comment]
> got the error message from octoprint:
> Changing monitoring state from "Operational" to "Offline (Error: SerialException: 'device reports readiness to read but returned no data (device disconnected or multiple access on port?)' @ comm.py:_readline:3831)"
> Connection closed, closing down monitor
> Changing monitoring state from "Offline" to "Detecting serial connection"
> Performing autodetection with 1 port/baudrate candidates: /dev/ttyUSB0@250000
> Trying port /dev/ttyUSB0, baudrate 250000
> Connecting to port /dev/ttyUSB0, baudrate 250000
> Handshake attempt #1 with timeout 2.0s
> Connected to: Serial<id=0x6a47d610, open=True>(port='/dev/ttyUSB0', baudrate=250000, bytesize=8, parity='N', stopbits=1, timeout=2.0, xonxoff=False, rtscts=False, dsrdtr=False), starting monitor

port already open in another instance for sure

**Comment by Snapperjagger** [comment]
maybe my webcam

**Comment by Snapperjagger** [comment]
bed:
Kp: +060.1
Ki: +008.07
Kd: +298.9
extruder:
kP: +024.6
kI: +002.90
kD: +052.3

**Comment by ils15** [comment]
> bed:
> Kp: +060.1
> Ki: +008.07
> Kd: +298.9
> extruder:
> kP: +024.6
> kI: +002.90
> kD: +052.3

thanks man
any advice?

**Comment by Snapperjagger** [comment]
powwer outage is off. You can enable on the screen but it will be nice to put it on for others

**Comment by Snapperjagger** [comment]
i will do a print test later. it's late in France. I leave you

**Comment by ils15** [comment]
> i will do a print test later. it's late in France. I leave you

no problem friend. we done a excellent job together
thank you

**Comment by Snapperjagger** [comment]
you're welcome. I will see later for the makerbase UI.

**Comment by Snapperjagger** [comment]
hey what's new!!
i just try the auto bed leveling and it work well in 3x3 points. Regarding the original firmware, when i executed an auto leveling, the hotend cames to the bed center at the end of the grid pointing and then go down to the z offset value. On the screen appeared the Z Offset setting. That permitted me to adjust the Z offset with a paper. With this procedure, it permit to start a print without doing any G29. now i can't as the hotend stay on the last point.
Do i need to proceed differently now? How can i adjust the Z offset now?
Other point, on the Probe Z offset (in motion menu) the plus and minus are reversed (when i touch plus it decrease the value and vice versa).
I will try a calibration cube for first print test.

**Comment by Snapperjagger** [comment]
first print ok
strangely the plus/minus issue disappear in the motion menu (if i remember well)

**Comment by Snapperjagger** [comment]
i have try to recompile with your last files but i got an error when compiling:
buildroot/share/PlatformIO/scripts/../../../../Marlin/src/inc/../../Configuration.h:2460:2: error: invalid preprocessing directive #TOUCH_ORIENTATION

**Comment by Snapperjagger** [comment]
missing #define before

File: config/examples/Two Trees/BlueR Plus/Configuration.h [hunk 1]
```diff
@@ -0,0 +1,2614 @@
...
+
+  // XPT2046 Touch Screen calibration
+	#define TOUCH_CALIBRATION_X 16128
+	#define TOUCH_CALIBRATION_Y 10841
+	#define TOUCH_OFFSET_X -13
+	#define TOUCH_OFFSET_Y -36
+  //#define TOUCH_ORIENTATION TOUCH_LANDSCAPE
+  
+
+#TOUCH_ORIENTATION TOUCH_PORTRAIT
```
Comment (Snapperjagger) [inline config/examples/Two Trees/BlueR Plus/Configuration.h]: missing #define before

**Pull request merged**